### PR TITLE
Add SYLT support

### LINKED ISSUE #44: Synchronized lyrics
Can I set synchronized lyrics? Or is it planned?

### PR DESCRIPTION
**Pull request by tognee**
Adds support for the "Synchronised lyrics" ID3 tag.
https://id3.org/id3v2.3.0#Synchronised_lyrics.2Ftext

This closes #44

### CODE REVIEW

**Comment by egoroof** [comment]
Did you find any media player (or other software) displaying SYLT frame?
I only found https://www.compuphase.com/software_sylteditor.htm, but it cannot read the frame generated from this branch (maybe due to not supporting multybite encoding).
I wonder how to test it externally

**Comment by tognee** [comment]
I used the python library mutagen as reference while implementing this field

File: src/ID3Writer.mjs [hunk 1]
```diff
@@ -482,6 +540,45 @@ export default class ID3Writer {
           offset += frame.value.byteLength;
           break;
         }
+        case 'SYLT': {
+          writeBytes = [1]; // encoding
+          writeBytes = writeBytes.concat(frame.language); // language
+          writeBytes = writeBytes.concat(frame.timestampFormat); // time stamp format
+          writeBytes = writeBytes.concat(frame.type); // content type
```
Comment (egoroof) [inline src/ID3Writer.mjs:547]: looks better?

```suggestion
          writeBytes = [1] // encoding
            .concat(frame.language) // language
            .concat(frame.timestampFormat) // time stamp format
            .concat(frame.type); // content type
```

File: src/ID3Writer.mjs [hunk 2]
```diff
@@ -296,6 +318,42 @@ export default class ID3Writer {
...
+        ) {
+          throw new Error('SYLT frame text value should be an array of pairs');
+        }
+        if (frameValue.type < 0 || frameValue.type > 6) {
+          throw new Error('Incorrect SYLT frame content type');
+        }
+        if (frameValue.timestampFormat < 1 || frameValue.timestampFormat > 2) {
+          throw new Error('Incorrect SYLT frame time stamp format');
+        }
+        frameValue.language = frameValue.language || 'XXX';
```
Comment (egoroof) [inline src/ID3Writer.mjs:345]: maybe let 'eng' be default value?

```suggestion
        frameValue.language = frameValue.language || 'eng';
```

**Review by egoroof — APPROVED** [review]

**Comment by egoroof** [comment]
could you rebase and resolve conflicts?

File: test/object/SYLT.mjs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      24, // size without header
+      0,
+      0, // flags
+      1, // text encoding
+      ...encodeWindows1252('eng'), // language
+      2, // Time stamp format
+      1, // Content type
+      0xff,
+      0xfe, // BOM
+      ...encodeUtf16le(''), // description (empty)
```
Comment (egoroof) [inline test/object/SYLT.mjs:42]: could you also add here not empty description to check if it is properly encoded?

**Comment by egoroof** [comment]
Ok, I've rebased & fix last review in https://github.com/egoroof/browser-id3-writer/pull/84 to be able to merge this.

Thank you!

**Pull request closed without merging**